src/EbmlBinary.cpp

```diff
@@ -97,7 +97,7 @@ filepos_t EbmlBinary::ReadData(IOCallback & input, ScopeMode ReadFully)
     return 0;
   }
 
-  Data = static_cast<binary *>(malloc(GetSize()));
+  Data = (GetSize() < SIZE_MAX) ? static_cast<binary *>(malloc(GetSize())) : nullptr;
```

Comment:
Can we please use `std::numeric_limits<std::size_t>::max` instead of macros such as `SIZE_MAX`? Not that I expect the values to differ, but `std::numeric_limits` is the proper C++ way to access such values. It also makes it much more explicit which type we're concerned about here.

Reply:
This is somehow correct but IMO not the right fit. We are not using a C++ allocator which would have it's matching C++ bounds. We are using the C `malloc()` which uses a size of `size_t` and has a maximum usable value of `SIZE_MAX`.
This is even used in this example describing size_t: https://en.cppreference.com/w/c/types/size_t

Whereas the value you mention may not be guaranteed to be there https://en.cppreference.com/w/cpp/types/numeric_limits
```
The standard library types that are aliases of arithmetic types (such as std::size_t or std::streamsize) may
also be examined with the std::numeric_limits type traits. 
```

I prefer to match the language parts together.

Reply:
Now, looking at the other parts needing `SIZE_MAX`, they use `new (std::nothrow)`.
I'll harmonize the code.

Reply:
Well, you could simply switch to [`std::malloc()`](https://en.cppreference.com/w/cpp/memory/c/malloc) which does take an `std::size_t` and which would be the C++ish thing to do.

When I wrote that comment the first time I was looking at `new char[…]`, and that one definitely uses `std::size_t`. I just added `dito` to the others without thinking much about `malloc`, to be honest.

Reply:
So we keep `malloc()` ans `SIZE_MAX` for this one ?
If libebml had some documentation that should be mentioned which one is allowed and which might be undefined behavior. And because of that I think we should stick to the C version to keep compatibility with how it worked so far.